src/components/Menus/Sidebar/InstructionsPanel/InstructionsPanel.jsx

```diff
@@ -79,13 +81,30 @@ const InstructionsPanel = () => {
     }
   }, [quizCompleted, currentStepPosition, numberOfSteps, dispatch, isQuiz]);
 
+  useEffect(() => {
+    dispatch(setProjectInstructions(editInstructions));
+  }, [editInstructions, dispatch]);
+
+  useEffect(() => {
+    if (project?.instructions) {
+      setEditInstructions(project.instructions);
+    }
+  }, [project, dispatch]);
+
   return (
     <SidebarPanel
       defaultWidth="30vw"
       heading={t("instructionsPanel.projectSteps")}
       Footer={ProgressBar}
     >
-      <div>{instructionsEditable && <p>Edit panel will go here</p>}</div>
+      <div>
+        {instructionsEditable && (
+          <textarea
+            value={editInstructions}
+            onChange={(e) => setEditInstructions(e.target.value)}
+          ></textarea>
+        )}
+      </div>
```

Comment:
Can we have a test for this, something like when text in textarea changes, correct action is fired?

Reply:
good shout, done in [1812e9e](https://github.com/RaspberryPiFoundation/editor-ui/pull/1163/commits/1812e9e7f851b2400ee3d4151d0c603034f7a150)